### PR TITLE
Fixes temporary block not applying to leap attacks.

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

This fixes furious steel by adding a check to see if parent returned FALSE (we already blocked) to catch cases where blocking is temporary

## Why It's Good For The Game

Dunno if this had other interactions; this fixes a weird interaction between body throws consuming the orbiting knives without actually stopping the hit.

## Changelog

:cl:
fix: temporary blocks such as blade heretic orbiting knives properly stop body throws.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -77,7 +77,7 @@
 		if(!hurt)
 			return
 
-		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK))
+		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK) || !.) // If parent was FALSE that means it got blocked
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon.dm:80]: I find this super sus

In throw code `. = TRUE` means "it was caught or blocked" and `. = FALSE` means it wasn't

On top of that shouldn't like, all of this code, cancel if it was "blocked"?

Comment (EnterTheJake) [reply code/modules/mob/living/carbon/carbon.dm:80]: If we block the code at the top the person being thrown doesn't get stunned.
And if it's not blocked it returns hitby.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Yeah this definitely ain't the right fix, the issue seems to stem that blocked attacks in `/mob/living/hitby` don't propagate down and return

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/mob/living/carbon/carbon.dm:80]: Testing this fix makes all body throws count as blocked as I anticipated

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -800,6 +801,6 @@
 
 /mob/living/proc/check_block(atom/hit_by, damage, attack_text = "the attack", attack_type = MELEE_ATTACK, armour_penetration = 0, damage_type = BRUTE)
 	if(SEND_SIGNAL(src, COMSIG_LIVING_CHECK_BLOCK, hit_by, damage, attack_text, attack_type, armour_penetration, damage_type) & SUCCESSFUL_BLOCK)
-		return TRUE
+		return SUCCESSFUL_BLOCK
```
Comment (MrMelbert) [inline code/modules/mob/living/living_defense.dm:804]: If you're changing this to be more bespoke about what's happening, you should also change `return FALSE` to `return NO_BLOCK` or some define (that you create)

Comment (MrMelbert) [reply code/modules/mob/living/living_defense.dm:804]: You also should update other uses of check block (silicons)

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 3]
```diff
@@ -86,7 +86,7 @@
 /mob/living/carbon/human/check_block(atom/hit_by, damage, attack_text = "the attack", attack_type = MELEE_ATTACK, armour_penetration = 0, damage_type = BRUTE)
 	. = ..()
 	if(.)
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/human/human_defense.dm:88]: And likewise, change `if(.)` to `if(. == SUCCESSFUL_BLOCK)`

File: code/modules/mob/living/living_defense.dm [hunk 4]
```diff
@@ -204,6 +204,7 @@
 			hitpush = FALSE
 			skipcatch = TRUE
 			blocked = TRUE
+			throwingdatum.blocked = TRUE
```
Comment (MrMelbert) [inline code/modules/mob/living/living_defense.dm:207]: This shouldn't be necessary - `hitby` is supposed to `return TRUE` if it was blocked, so as I mentioned before, it should be returning `TRUE` here.

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_living.dm [hunk 5]
```diff
@@ -244,6 +244,7 @@
 
 ///from /mob/living/proc/check_block(): (atom/hit_by, damage, attack_text, attack_type, armour_penetration, damage_type)
 #define COMSIG_LIVING_CHECK_BLOCK "living_check_block"
+	#define FAILED_BLOCK FALSE
```
Comment (MrMelbert) [inline code/__DEFINES/dcs/signals/signals_mob/signals_mob_living.dm:247]:
```suggestion
	#define FAILED_BLOCK NONE
```

File: code/modules/mob/living/carbon/carbon.dm [hunk 6]
```diff
@@ -77,7 +77,7 @@
 		if(!hurt)
 			return
 
-		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK))
+		if(victim.check_block(src, 0, "[name]", LEAP_ATTACK) || . == SUCCESSFUL_BLOCK)
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon.dm:80]: I feel like we should check that parent call was blocked before running check block, so we don't needlessly call check block if we know it already is 
```suggestion
		if(. == SUCCESSFUL_BLOCK || victim.check_block(src, 0, "[name]", LEAP_ATTACK))
```

File: code/modules/mob/living/living_defense.dm [hunk 7]
```diff
@@ -204,10 +204,11 @@
 /mob/living/hitby(atom/movable/AM, skipcatch, hitpush = TRUE, blocked = FALSE, datum/thrownthing/throwingdatum)
 	if(!isitem(AM))
 		// Filled with made up numbers for non-items.
-		if(check_block(AM, 30, "\the [AM.name]", THROWN_PROJECTILE_ATTACK, 0, BRUTE))
+		if(check_block(AM, 30, "\the [AM.name]", THROWN_PROJECTILE_ATTACK, 0, BRUTE) == SUCCESSFUL_BLOCK)
```
Comment (MrMelbert) [inline code/modules/mob/living/living_defense.dm:207]:
```suggestion
		if(check_block(AM, 30, "\the [AM.name]", THROWN_PROJECTILE_ATTACK, 0, BRUTE) & SUCCESSFUL_BLOCK)
```

**Pull request merged**